### PR TITLE
Feature/collateral pool wrapper #69

### LINKED ISSUE #69: Create class to wrap MarketCollateralPool objects
### Description
*Type*:  Feature
We have created wrappers for many of the main contract objects so they are persistent, but have not done so yet for the MarketCollateralPool 

### Solution
Possibly to create a class that wraps both a MarketContract and the MarketCollateralPool together into the same object, or a MarketCollateralPoolWrapper that stores all of the collateral pools separately from all of their corresponding MarketContracts.  Either could work, and have small differences in design and implications 

### Definition of Done
- Create persistent objects for MarketCollateralPool
- Refactor Collateral.ts to use these new objects.

### PR DESCRIPTION
**Pull request by pelsasser**
##### Description
This is a bit of a big one, but will simplify everyone's life greatly.  No more worrying about which address to use, just always use the MarketContract address, and all the functionality works as expected.

##### Checklist
- [x] Linter status: 100% pass 
- [x] Changes don't break existing behavior - not in this repo ;)
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
Just about everything

##### Refers/Fixes
Fixes #69

### CODE REVIEW

**Comment by pelsasser** [comment]
Okay calling this done, if someone wants to take up a review.  @rojosnow @eswarasai whomever has time would be great.  Thanks! I am sure there will be questions / discussion on it.

**Comment by pelsasser** [comment]
also @perfectmak if you have any input would love this as well, this is probably gonna cause you some headache on what your working on.

**Comment by perfectmak** [comment]
@pelsasser Wow! Indeed a big one. Yeah, it would cause me some headache, but it should be straight forward to fix.

File: src/contract_wrappers/MarketProtocolContractWrapper.ts [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+import { constants } from '../constants';
+import { createOrderHashAsync, isValidSignatureAsync } from '../lib/Order';
+import { OrderTransactionInfo } from '../lib/OrderTransactionInfo';
+import { assert } from '../assert';
+import { MarketProtocolContractSetWrapper } from './MarketProtocolContractSetWrapper';
+import { Market } from '../Market';
+import { Transaction } from '@0xproject/types';
+const Decoder = require('ethereum-input-data-decoder');
+
+export interface CollateralEvent {
```
Comment (eswarasai) [inline src/contract_wrappers/MarketProtocolContractWrapper.ts]: Need to move this to `types` folder

**Pull request merged**